### PR TITLE
change: adjustments to support ImageRoleBindings in Manager

### PR DESCRIPTION
**Pull request by iwilltry42**
This PR includes some refactoring to expose functionality to manager to use with the new ImageRoleBindings feature.

### Checklist
- [ ] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [ ] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/runtime/pull/1199)
- [ ] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [ ] Commits follow [contributing guidance](https://github.com/acorn-io/runtime/blob/main/CONTRIBUTING.md#commits)
- [ ] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [ ] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: integration/client/signatures/rules_test.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func TestImageAllowRules(t *testing.T) {
 	assert.Error(t, err, "should error since image is not covered by images scope of IAR")
 
 	// update image allow rule to cover that image
-	iar.Images = []string{tagName}
+	iar.Images = []string{"**"}
```
Comment (g-linville) [inline integration/client/signatures/rules_test.go:123]: Why this change?

File: pkg/imageallowrules/imageallowrules.go [hunk 2]
```diff
@@ -106,73 +106,74 @@ func CheckImageAgainstRules(ctx context.Context, c client.Reader, namespace stri
...
 			continue
 		}
 
 		// > Signatures
 		// Any verification error or failed verification issue will skip on to the next IAR
 		for _, rule := range imageAllowRule.Signatures.Rules {
-			if err := cosign.EnsureReferences(ctx, c, image, namespace, &verifyOpts); err != nil {
-				logrus.Infof("failed checking image %s against %s/%s.signatures: %v", image, imageAllowRule.Namespace, imageAllowRule.Name, err)
+			if err := VerifySignatureRule(ctx, c, namespace, image, rule, verifyOpts); err != nil {
+				logrus.Errorf("Verification failed for %s against %s/%s: %v", image, imageAllowRule.Namespace, imageAllowRule.Name, err)
```
Comment (g-linville) [inline pkg/imageallowrules/imageallowrules.go:118]: Imo the `error` level seems to be a bit too severe for this. If I were scanning through logs, I would think that something had gone wrong with Acorn itself, when really the user was just trying to verify an image with the wrong signature or something. Imo `info` or `warn` would be better for this.

**Review by g-linville — APPROVED** [review]

File: integration/client/signatures/rules_test.go (hunk 1 above)
Comment (iwilltry42) [reply integration/client/signatures/rules_test.go:123]: I will merge that change away with https://github.com/acorn-io/runtime/pull/2113 :+1:

File: pkg/imageallowrules/imageallowrules.go (hunk 2 above)
Comment (iwilltry42) [inline pkg/imageallowrules/imageallowrules.go:118]: Oh you're right. That wasn't meant to stay.
```suggestion
				logrus.Infof("Verification failed for %s against %s/%s: %v", image, imageAllowRule.Namespace, imageAllowRule.Name, err)
```

Comment (iwilltry42) [reply pkg/imageallowrules/imageallowrules.go]: Absolutely! Updated.
